k8s/envoy-x509/README.md

```diff
@@ -219,93 +210,93 @@ We have deployed our services and also registered them in SPIRE. Now we can chec
+Wait some seconds for the deployment to propagate before trying to view the `frontend-2` service in your browser again.
+Once the pod is ready, refresh the browser using the correct URL for `frontend-2` service (e.g. `http://35.222.190.182:3002`). As a result, now Envoy does not allow the request to get to the `backend` service and account details are not shown in your browser.
 
 ![Frontend-2-no-details][frontend-2-view-no-details]
 
 [frontend-2-view-no-details]: images/frontend-2_view_no_details.png "Frontend-2 view no details account"
 
-
-On the other hand, you can check that the **frontend** service is still able to get a response from the **backend**. Refresh the browser at the correct url, in this case `35.222.164.221:3000`, and confirm that user details for _Jacob Marley_ are displayed.
+On the other hand, you can check that the `frontend` service is still able to get a response from the `backend`. Refresh the browser at the correct URL (e.g. `http://35.222.164.221:3000`) and confirm that account details are shown for _Jacob Marley_.
 
 
 # Cleanup
 
-When you are finished running this tutorial, you can use the following commands to remove all the resources used for Envoy integration and the SPIRE setup.
+When you are finished running this tutorial, you can use the following script to remove all the resources used for configuring Envoy to perform X.509 authentication on workload's behalf. This command will remove:
+   * - All resources created for the SPIRE - Envoy X.509 integration tutorial.
+   * - All deployments and configurations for the SPIRE agent, SPIRE server, and namespace.
 
-## Kubernetes Cleanup
+```console
+   $ bash scripts/clean-env.sh
+```
 
-Keep in mind that these commands will also remove the setup that you configured in the [Kubernetes Quickstart](/spire/try/getting-started-k8s/).
 
-1. Delete all resources created for this SPIRE - Envoy integration tutorial:
+# Extend the Scenario with a Role Based Access Control filter
 
-   ```console
-   $ kubectl delete -k k8s/.
-   ```
+Envoy provides a Role Based Access Control (RBAC) HTTP filter that checks the request based on a list of policies. A policy consists of a list of permissions and principals where the principal specifies the downstream client identities of the request, for example, the URI SAN of the downstream client certificate. So we can use the SPIFFE ID assigned to the service to create policies that allows a more granular access control.
 
-2. Delete all deployments and configurations for the SPIRE agent, SPIRE server, and namespace:
+For example, it is possible to create a policy that allows the `frontend` service to obtain only data for profiles. This is achieved by defining a policy which principal matches the SPIFFE ID of the service and the permissions allows only GET requests to the `/profiles` resource.
```

Comment:
I think we need to define what a "profile" is here. Also because of that, I have no idea if my change to the first sentence makes sense. ❓ 

```suggestion
For example, it is possible to create a policy that allows the `frontend` service to only read data from profiles. This is achieved by defining a policy in which the principal matches the SPIFFE ID of the service and the permissions allow only GET requests to the `/profiles` resource.
```

Reply:
The web app use three endpoints to get the all the data that is displayed. One of those resources returns the details about the owner of the bank account. That is the /profile resource. I'll add some text to clarify this.
Thanks for pointing it out Steve.